Implementation for hangman using OO in python

bahetiNaman opened the pull request.

bahetiNaman committed:
feature: basic youtube app with dark/light theme

bahetiNaman committed:
fix: removed some unnecessary images and logo files

bahetiNaman committed:
fix: removed some unnecessary images and logo files

bahetiNaman committed:
fix: fixed the issue of different height of searchInput and searchButton in navBar

bahetiNaman committed:
fix: showing full video description on hover

bahetiNaman committed:
Feature: Initial implementation for hangman game

Implemented hangman using OO in python. Functionality includes:
 - created a class for hangmanGame.
 - created a class for wordGenerator that gives random words from file 'sowpods.txt'.
 - created a class for Hangman which updates its structure based on number of incorrect guesses.

gekco commented on Hangman/HangmanGame.py:
```diff
@@ -0,0 +1,119 @@
+
+        want_to_restart = (input("To restart, enter Yes: (case insensitive) ")).lower()
+        if want_to_restart == "yes":
+            self.__init__()
```
we are caling the constructor inside the object method itself ? in this case it will work but in general you __init__ does a lot more, ideally there either should be a reset function or a class function which creates/manages multiple objects and multiple play by calling object.single_play()

gekco commented on Hangman/HangmanGame.py:
```diff
@@ -0,0 +1,119 @@
+        else:
+            self.status = 'guessing'
+
+    def get_status(self) -> str:
```
not required

gekco commented on Hangman/HangmanGame.py:
```diff
@@ -0,0 +1,119 @@
+        else:
+            self.incorrect_guess_count += 1
+            self.hangman.on_incorrect_guess()
+        self.__set_status()
```
why __ ?

bahetiNaman replied:
Only HangmanGame class is using this set_status and any other class should not be able to change game status. So, it should be private right?

gekco commented:
Looks good. just a few minor changes. good work

sukhdeep4395 commented on Hangman/HangmanGame.py:
```diff
@@ -0,0 +1,119 @@
+        self.status = ''
+
+    def start_game(self) -> None:
+        self.__set_word(self.word_generator.get_next_word())
```
start_game is doing too many things, we should abstract it out further. 
like 17 and 18 can be moved to self.initialize_game()

sukhdeep4395 commented on Hangman/Hangman.py:
```diff
@@ -0,0 +1,53 @@
+class Hangman:
```
we can better structure this project. 
folders like models, constants, tests etc.

bahetiNaman committed:
fix: added changes mentioned in the code review
main.py: added WordGenerator and Hangman as arguments in HangmanGame
HangmanGame.py: start_game was doing too much work. Broke it into reset_game, end_game and run_game
Hangman.py: created reset_hangman function
WordGenerator.py: previously opened file was not being closed. Fixed that.

bahetiNaman committed:
Revert "fix: removed some unnecessary images and logo files"

This reverts commit e13765ed037e55838845431792a4f3b241734796.

bahetiNaman committed:
Revert "feature: basic youtube app with dark/light theme"

This reverts commit c8a4f065a21279023175bb0e389b2b27425899d1.

bahetiNaman committed:
Revert "fix: removed some unnecessary images and logo files"

This reverts commit ea0c05e436b46914a0a730907c5e737c3fa46e9b.

bahetiNaman committed:
Revert "fix: fixed the issue of different height of searchInput and searchButton in navBar"

This reverts commit 2cc1e8cdd3c185c494f754deeebfb51e25a57345.

bahetiNaman committed:
Revert "fix: showing full video description on hover"

This reverts commit 64d7e2871747063424d8f3edbd0974532e5c6a06.
